### ISSUE TITLE
Refactor the Assays/Attachment feature

### ISSUE THREAD

**Issue by diegoliberman** [instedd/cdx#1331]

**Comment by straight-shoota** [comment]
Could someone (@devduarte ?) fill in the reasons for splitting `AssayFile` as a separate model from `AssayAttachment`? It's just a single-field data type with a 1:1 relation to `AssayAttachment`. As far as I can see, there's no apparent reason why this couldn't have stayed all in a single model.

**Comment by devduarte** [comment]
It was just a technical decision to make our lives a bit easier with this feature. The firsts versions of `AssayAttachment` included the file attribute, as we implemented Dropzone we needed to append the files data to the form but that implied more (unnecessary) work at that time since we needed to create the whole form a Dropzone and make other tweaks to make it work. Also, since we were working with only one file input (can't recall why, now) when user wanted to upload files in two different events, the files from the first event got overwritten. Handling the files async, as Dropzone allowed us to do that, felt like the answer we were looking for. 
Having two separate models is a consecuence of the async file handling. If the user uploaded a file, we would create an `AssayAttachment` but if the browser tab got closed the AssayAttachment would have been created without a parent Sample entity. So we created `AssayFile` to handle this behaviour separately from `AssayAttachment` and `Sample`.
